### ISSUE TITLE
macOS - build for ARM on Intel machine?

### ISSUE THREAD

**Issue by emendelson** [joncampbell123/dosbox-x#2789]
Is it possible to build DOSBox-X for an ARM Mac on an Intel machine running Big Sur? Before I try experimenting with -arch settings (and almost certainly failing), I thought it was worth asking here. It's now possible to build ARM versions of the BasiliskII and SheepShaver emulators on Intel, and I hope it might be possible to do the same with DOSBox-X. But I know that this may be only a hope.

**Comment by Torinde** [comment]
@maron2000 @JoeBroesele, can you make automatic building of macOS arm64 portable? Currently it's present in [releases](https://github.com/joncampbell123/dosbox-x/releases/tag/dosbox-x-v2023.05.01), but not in [nightly builds/actions artifacts](https://github.com/joncampbell123/dosbox-x/actions/workflows/macos.yml).

**Comment by maron2000** [comment]
I need to know whether there is a trivial way to cross compile binaries for M[1-3] Mac on an intel platform.
Until I or someone else figures it out, we need someone to build and upload it.

**Comment by Torinde** [comment]
Will that help [Cross-compilation is possible (and was tested) in both directions, i.e. producing arm64 binaries on x86_64 host](https://github.com/qbittorrent/qBittorrent/wiki/Compilation:-macOS-(x86_64,-arm64,-cross-compilation))?

**Comment by emendelson** [comment]
I once tried to figure this out. It's not at all trivial, and I stopped trying. What makes it so difficult is that all the different components need to be built cross-platform, and the syntax is different for each one. The page about qbittorrent uses cmake, and all the DOSBox-X build scripts would have to be rewritten for it. This would be a major project for someone with an enormous amount of skill. Definitely not trivial.

**Comment by Torinde** [comment]
Staging has both [ARM64 and universal artifacts](https://github.com/dosbox-staging/dosbox-staging/actions/runs/5633722097)
![image](https://github.com/joncampbell123/dosbox-x/assets/97228894/f2102793-8e8d-4188-9386-46afe08adbb4)

**Comment by maron2000** [comment]
It's runnuing on a self-hosted runner so it doesn't help.
Of course you can contribute by providing one.

**Comment by Torinde** [comment]
Clear, didn't know that you can do this automatic github builds on your own computer. 
I don't have such hardware, :(

**Comment by maron2000** [comment]
Yep, it's great that dosbox-staging prepared a building environment.
Still, ARM Mac users can build nightly versions on their own, it's shouldn't be so difficult.

**Comment by maron2000** [comment]
A test to add nightly ARM mac build.
I can't test it myself, so I don't know whether it works, but at least succeeded in building it.
https://github.com/maron2000/dosbox-x-1/actions/runs/8243795766

**Comment by emendelson** [comment]
@maron2000 - Yes, it runs under ARM Sonoma. Haven't tried under earliler versions. Thank you!
